feat(Webflow Node): Update to use the v2 API

Joffcom opened the pull request:
## Summary
Updates the Webflow node to use the v2 API.

Tests to follow

## Related Linear tickets, Github issues, and Community forum posts
https://github.com/n8n-io/n8n/issues/7052
https://community.n8n.io/t/webflow-integration-isn-t-working/48185/

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

Joffcom committed:
Initial migration work

Joffcom committed:
fix get item

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into node-761-webflow-update-to-v2-api

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into node-761-webflow-update-to-v2-api

Joffcom committed:
Last files

ShireenMissi commented on packages/nodes-base/nodes/Webflow/V1/GenericFunctions.ts:
GenericFunctions should remain in the level above the version so they can be used by any/all versions

Joffcom replied:
When we version a new node like this and move it to the new directory structure the generic functions gets moved to transport, This also allows us to make changes to the transport without the risk of breaking the older version that users may still be using. It also allows other parts to continue using the old method if needed like the Trigger node, 

We could in theory put generic functions at the top level then copy / paste the functions or add a parameter for node version but it carries more risk, It is up to you what we do here but we will need to make sure that we follow this for all new node overhauls.

netroy replied:
We definitely have an inconsistent setup across the package, and `GenericFunctions.ts` has also become a dumping ground. 
We should discuss this in the nodes team, vote on it, and then figure out a way to enforce this.
We should also figure out better way to share code between versions that ensures that the common code does not implement **any** version specific logic.

In my personal opinion however, `GenericFunctions.ts` should always be version-**generic**, not version-specific, and should definitely not be in a version folder.
 
Putting a file named `GenericFunctions.ts` in a version folder is like calling a file `SharedFunctions.ts`, but importing it only in one single place 😅

Joffcom replied:
Sounds like I am moving it out then 😄

ShireenMissi commented on packages/nodes-base/nodes/Webflow/V2/methods/index.ts:
```diff
@@ -0,0 +1,2 @@
+//export * as listSearch from './listSearch';
```
what's the reason for having this comment?

Joffcom replied:
Ah I was toying with the idea of adding more features to make the node better but then realised I should start with a 1:1 copy and we wait for user feedback on new features.

ShireenMissi commented on packages/nodes-base/nodes/Webflow/WebflowTrigger.node.ts:
```diff
@@ -9,7 +9,7 @@ import type {
 } from 'n8n-workflow';
 
-import { webflowApiRequest } from './GenericFunctions';
+import { webflowApiRequest } from './V1/GenericFunctions';
```
related to the first comment, I think we should avoid referencing files within the version folder

Joffcom replied:
This is referenced while I look at the Trigger node in another PR as it could be trickier to move depending on how Node versioning is at the moment with trigger nodes, Last time I tried it the versioning wasn't working so this would at least keep the trigger node working until we get a chance to fix it.

ShireenMissi reviewed (commented):
While manual testing this it seems that V2 doesn't have the right credentials set up

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into node-761-webflow-update-to-v2-api

Joffcom committed:
review fixes

Joffcom commented:
@ShireenMissi ready for round 2, Removed the oauth selection as that isn't needed so we have the only credential to select and removed the comment for the feature I didn't add

Joffcom committed:
Review tweaks

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into node-761-webflow-update-to-v2-api

Joffcom committed:
Move loadOptions to genericFunctions and add v2 Trigger node

ShireenMissi commented on packages/nodes-base/credentials/WebflowOAuth2Api.credentials.ts:
```diff
@@ -16,6 +16,13 @@ export class WebflowOAuth2Api implements ICredentialType {
+			name: 'legacy',
+			type: 'boolean',
+			default: true,
+			description: 'If the legacy API should be used',
```
when manually testing with legacy option on I get "
403 - {"message":"OAuthForbidden: You are missing the following scopes - 'cms:read'","code":"missing_scopes","externalReference":null,"details":[]}"

Joffcom replied:
That makes sense the legacy option is only for the v1 node as we don't have credential versioning yet.

I will update the wording to make it clearer.

janober commented:
Got released with [`n8n@1.54.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.54.0)
